src/states/node.rs

```diff
@@ -1426,6 +1426,17 @@ impl Node {
             }
         }
 
+        let acting_as_proxy = if let Authority::Client { ref proxy_node_name, .. } =
+                                     routing_msg.src {
+            proxy_node_name == self.name()
+        } else {
```

Comment:
It would be more concise to remove the `else` branch, and just have the `if ... { return Ok(()); }` in the `if let` branch. That would also avoid some unnecessary calls to `is_in_our_group`.